tools/shared/src/main/scala/org/scalajs/core/tools/javascript/ScalaJSClassEmitter.scala

```diff
@@ -66,6 +69,20 @@ final class ScalaJSClassEmitter private (
     linkingUnit.classDefs.map(c => c.encodedName -> c).toMap
   }
 
+  private[javascript] lazy val hasSubclasses: Set[String] = {
+    if (linkingUnit == null) {
+      throw new IllegalArgumentException(
+          "A class emitter created without a LinkingUnit cannot emit JS classes")
```

Comment:
If you drop the deprecated constructors, yes, you should drop all the `linkingUnit == null` tests that you can find.

The deprecated constructors were useful to preserve backwards binary compatibility of the Tools API wrt to earlier versions of Scala.js. But since we are officially breaking compatibility of the Tools between 0.6.5 and 0.6.6 anyway, they can go away.